src/placeos-models/user.cr

```diff
@@ -85,11 +85,17 @@ module PlaceOS::Model
     # Callbacks
     ###############################################################################################
 
+    before_destroy :ensure_user_remains
     before_destroy :cleanup_auth_tokens
     before_save :build_name
     before_save :create_email_digest
 
-    # deletes auth tokens
+    # Prevent the system from entering state with no users
+    protected def ensure_user_remains
+      raise Model::Error.new("At least one user must remain") unless User.count > 1
```

Comment:
Do we need to worry about concurrent destroys here?

Reply:
I have implemented db locks for rethinkORM, so we could use that. Might leave that as a todo for now, and solve it in another PR.